### ISSUE TITLE
Analyse code that has Lombok as dependency

### ISSUE THREAD

**Issue by staleks** [autonomousapps/dependency-analysis-android-gradle-plugin#188]
Hi,

I have Java based Gradle Project that has multiple sub-projects (java too). 
"Master" Gradle Project is just a wrapper and doesn't have any line of Java code.

Subprojects "heavily depend" on using lombok.

```
    // lombok
    compileOnly "org.projectlombok:lombok:${lombokVersion}"
    annotationProcessor "org.projectlombok:lombok:${lombokVersion}"
```

Gradle version: 6.4
Java version: 11
id "com.autonomousapps.dependency-analysis" version "0.46.0"

After I've added this plugin, I was able to execute: 
```
./gradlew buildHealth 
```
and got advice-*.json's as result of report.

But now if I try to run
```
./gradlew clean build
```
with this plugin, I'm getting following error:

> Task :app:findDeclaredProcsMain
Could not reflectively access processor class lombok.launch.AnnotationProcessorHider$AnnotationProcessor

> Task :app:abiAnalysisMain FAILED
> Task :app:analyzeClassUsageMain FAILED

FAILURE: Build completed with 2 failures.

And failures are:
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:abiAnalysisMain'.
> A failure occurred while executing com.autonomousapps.tasks.AbiAnalysisWorkAction

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:analyzeClassUsageMain'.
> There was a failure while executing work items
   > A failure occurred while executing com.autonomousapps.tasks.JarAnalysisWorkAction

Any help/answer would be appreciated.

Thank You

**Comment by autonomousapps** [comment]
Would you be able to provide a build scan from the failure?

**Comment by staleks** [comment]
I'll try to prepare "dummy" project so you can test it your own. Will let you know here once I'm done.

**Comment by autonomousapps** [comment]
Thank you, a reproducer would be very helpful. Would love to get this resolved for you.

**Comment by autonomousapps** [comment]
I took a stab at this myself, but was unable to reproduce an issue. A reproducer would be great.

**Comment by staleks** [comment]
Hi,
I've prepared you non-working (master branch) and working (develop branch) example.

You can find it here: 
https://github.com/staleks/lombok-analyse-android

If you need any more details please let me know.

**Comment by autonomousapps** [comment]
Thank you, I expect that will be very helpful.

**Comment by autonomousapps** [comment]
I've taken a look, and I'm not sure this has anything to do with lombok. It _may_ instead be something to do with the spring plugin. Take a look at this [build scan](https://scans.gradle.com/s/xpvz4yyxpk6xc/timeline?details=tmjprnevdioz2), and note that `app:jar` is SKIPPED. I ran another build with `-i` and saw that it was skipped because it's `onlyIf` is `false` (i.e., the jar task is disabled). The plugin depends on that task for various kinds of analysis.

Before I dive any deeper, I think I need to understand why the jar task is being disabled. I don't have any experience with the spring framework. Can you shed any light?

**Comment by staleks** [comment]
Maybe this will shed some light?
https://stackoverflow.com/a/49361992/1437865

**Comment by staleks** [comment]
As suggested by that post, I've explicitly added in app/build.gradle
```
jar {
     enabled = true
}
```
Please check following branch: feature/DAA-188-analysis
URL: https://github.com/staleks/lombok-analyse-android/tree/feature/DAA-188-analysis

Now it works :)

Thank You for all of your effort.

Now if I look output, I've noticed following statement:
```
> Task :domain:advicePrinterMain
Unused annotation processors that should be removed:
- compileOnly("org.projectlombok:lombok:1.18.12")
```
But as you can notice, I've added some Lombok's annotations on Country.java class.

Can you please elaborate on this?

**Comment by autonomousapps** [comment]
Thanks for confirming that enabling the jar task solves it.

Regarding that advice, it looks like a bug! I'll take a look.

**Comment by autonomousapps** [comment]
So, one of the things the plugin does to detect unused annotation processors is look at the imports in the source code and compare that against the declared supported annotation types of the processor. Lombok declares that it supports `lombok.*`, but the source imports `lombok.AccessLevel`, `lombok.Getter`, and `lombok.NoArgsConstructor`. tl;dr I need to support star notation :D

**Comment by autonomousapps** [comment]
I now have a PR fixing the star-import issue, but it doesn't address the disabled jar issue. Is that a blocker for you? Trying to handle that would be ... awkward.

**Comment by staleks** [comment]
Hi,

as long as you have it somewhere in documentation, for this combination, I'm ok as we "fixed" it :)

**Comment by autonomousapps** [comment]
Sounds good! I'll update the docs. Thanks for the issue and working with me to resolve it.

**Comment by autonomousapps** [comment]
I may have a slightly better solution than adding this to the docs. Looks like I can check if the jar task is enabled in the `onlyIf` predicate of a consuming task, and then emit a more specific warning.

**Comment by autonomousapps** [comment]
I just released v0.49.0 which resolves this.
